pkg/providers/amifamily/ami.go

```diff
@@ -117,18 +117,18 @@ func NewDefaultProvider(versionProvider version.Provider, ssm ssmiface.SSMAPI, e
 
 // Get Returning a list of AMIs with its associated requirements
 func (p *DefaultProvider) Get(ctx context.Context, nodeClass *v1beta1.EC2NodeClass, options *Options) (AMIs, error) {
-	var err error
-	var amis AMIs
-	if len(nodeClass.Spec.AMISelectorTerms) == 0 {
-		amis, err = p.getDefaultAMIs(ctx, nodeClass, options)
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		amis, err = p.getAMIs(ctx, nodeClass.Spec.AMISelectorTerms)
+	amis, err := p.getAMIs(ctx, nodeClass.Spec.AMISelectorTerms)
+	if err != nil {
+		return nil, err
+	}
+	if term, ok := lo.Find(nodeClass.Spec.AMISelectorTerms, func(term v1beta1.AMISelectorTerm) bool {
+		return term.EKSOptimized != nil
+	}); ok {
+		optimizedAMIs, err := p.getOptimizedAMIs(ctx, term.EKSOptimized.Family, options)
```

Comment:
IMO the two functions go about getting AMIs in very different ways and make sense to keep separate. I could see renaming `getAMIs` to be more specific since it currently implies get all AMIs. Having trouble thinking of a good name though, maybe `getFilterAndOwnerSetAMIs`?

Reply:
That's fair, I'm happy to keep as separate calls. Personally I think having the check in the order how it previously was makes a little bit more code sense to me. If I'm reading line by line, I'd personally be surprised that p.getAMIs() would be succeeded by another type of GetAMI call. Up to you though.